### ISSUE TITLE
 Update Categories List

### ISSUE THREAD

**Issue by rgaudin** [kiwix/overview#104]
[was originally incorrectly opened at https://github.com/offspot/overview/issues/5]

As discussed during Hackathon, we acknowledge that the current list of Categories (source-website based) is not satisfactory.

We want to replace it with a new –limited– list, based on _Knowledge families_ ; inspired by the Dewey Classification of the root categories of The Library of Congress or others. @Popolechien will submit something.

We also want those categories to be translated which will lead to a libkiwix ticket.
We also want to explore iconography for categories but this discussion will happen once we have settled on the new list.

---

@kelson42 wrote on May 3 2024, 2:08 PM GMT

> _Disclaimer: This is an ongoing draft_
> 
>     * Literature (Romans, books, essays, ...)
> 
>     * Conferences
> 
>     * News (Kind of "journalism" work)
> 
>     * Courses
> 
>     * Knowledge References (Encyclopedia, dictionnary, scientific publication, ...)
> 
>     * Popular culture (Fan work)
> 
>     * Kids (dedicated to kids <12)
> 
>     * Know-how Tutorial & guides (howtos of any sort, technical docs)
> 
>     * Vulgarisation (explaining anything in a simple manner outside proper school/curriculum)
> 
>     * Other

@Popolechien wrote on May 8 2024, 1:38 PM GMT

> I was going to suggest working off [Dewey classes](https://en.wikipedia.org/wiki/List_of_Dewey_Decimal_classes) or the [Library of Congress Classification](https://en.wikipedia.org/wiki/Library_of_Congress_Classification), but considering the type of content we have this might work better.

**Comment by benoit74** [comment]
Next steps discussed today:
- @Popolechien  proposes an alternative list of categories (set 2), more or less based on Library of Congres
- @kelson42 confirms its draft of categories (set 1) is OK as a first version
- @benoit74 extracts a random set of 50 ZIMs to classify in terms of categories and tags, and create a spreadsheet to fill with the two alternatives (set 1 as category and set 2 as tags ; set 2 as category and set 1 as tags)
- everyone interested does a proposal of classification / tagging

**Comment by benoit74** [comment]
Spreadsheet is ready at https://docs.google.com/spreadsheets/d/12pLOEERVqrPtAU7yc6NoIn3bBUDh5Hz3vjB7bQFIwRs/edit?usp=sharing

I selected only 30 ZIMs because it was already a significant work to find them, and probably as well a significant work to fill the sheet. Feel free to add some if you miss some.

It now misses the two list of categories from @Popolechien and @kelson42

**Comment by Popolechien** [comment]
Yeah I've asked librarians about it and they should get back to me soon-ish. Apparently my question was more complex than I anticipated.

**Comment by benoit74** [comment]
On a side note, and as mentioned in https://github.com/kiwix/kiwix-apple/issues/1023, we need to decide if the category is a technical label (without space, ..., e.g. `stack_exchange` as today) and if we have "something / somewhere" which allows readers to know how to transform it into a nice looking label (e.g. `StackExchange`), or if we directly store the nice looking label.

From my perspective, I see only advantages to storing directly the nice looking label as category. 

Note that same question applies to tags.

And it also makes me realize that we need to open the question of how we handle the transition from old (current) categories to new categories. We know that there will be ZIMs with old categories in the wild for a long time, but do we try at least to fix all categories at once in the online library, meaning we fix the Zimfarm recipes and we recreate all existing ZIMs with fixed categories? And tags obviously.

**Comment by rgaudin** [comment]
Sure hope we wont have a StackEchange category! 😉

As asked, the question sounds silly but the underlying reason for an _ID_ and not a label is that we (I believe) settled on having categories localized (contrary to tags). ID would allow smoother updates to both English and translated labels.

Needs to be discussed and documented obviously

**Comment by benoit74** [comment]
All this makes sense, thank you for remembering what I easily forget (including the StackExchange remark 🤣)

**Comment by Popolechien** [comment]
> From my perspective, I see only advantages to storing directly the nice looking label as category.

I concur.

as for categories / tags update, that's going to be a long ass effort, but if it is not doable via script then it would be best for the person coming up with the category tree to do the first pass IMHO
